src/k8s-engine/k3sHelper.ts

```diff
@@ -35,6 +38,98 @@ import { showMessageBox } from '@/window';
 
 const console = Logging.k8s;
 
+// kubernetes-client/javascript doesn't have support for the `proxy-url` cluster field.
+// We are providing variants of loadFromString() and exportConfig() that do.
+
+interface Cluster {
+  readonly name: string;
+  readonly caData?: string;
+  caFile?: string;
+  readonly server: string;
+  readonly skipTLSVerify: boolean;
+  readonly tlsServerName?: string;
+  readonly proxyUrl?: string;
+}
+
+function loadFromString(kubeConfig : KubeConfig, config: string, opts?: Partial<ConfigOptions>): void {
+  const obj = yaml.parse(config) as any;
+
+  kubeConfig.clusters = newClusters(obj.clusters, opts);
+  kubeConfig.contexts = newContexts(obj.contexts, opts);
+  kubeConfig.users = newUsers(obj.users, opts);
+  kubeConfig.currentContext = obj['current-context'];
+}
+
+export function newClusters(a: any, opts?: Partial<ConfigOptions>): Cluster[] {
```

Comment:
I don't think we need to export this function if it's only used locally.

Reply:
Yeah, I just copied it verbatim from upstream; will remove.